spec/support/command_helpers.rb

```diff
@@ -1,17 +1,270 @@
+
+    app = prefix
+    app += "-#{suffix}" unless suffix.nil? || suffix.empty?
+    return app unless create_if_not_exists
+
+    params = CREATE_APP_PARAMS[extra_prefix] || CREATE_APP_PARAMS["default"]
+    create_app_if_not_exists(app, **params)
+
+    app
+  end
+
+  def apps_to_delete
+    @@apps_to_delete ||= [] # rubocop:disable Style/ClassVars
+  end
+
+  def create_app_if_not_exists(app, deploy: false, image_before_deploy_count: 0, image_after_deploy_count: 0) # rubocop:disable Metrics/MethodLength
+    apps_to_delete.push(app)
+
+    result = run_cpl_command("exists", "-a", app)
+    return app if result[:status].zero?
+
+    puts "\nCreating app '#{app}' for tests\n\n" if ENV.fetch("VERBOSE_TESTS", nil) == "true"
+
+    run_cpl_command!("setup-app", "-a", app, "--skip-secret-access-binding")
+
+    image_before_deploy_count.times do
+      run_cpl_command!("build-image", "-a", app)
+    end
+    run_cpl_command!("deploy-image", "-a", app) if deploy
+    image_after_deploy_count.times do
+      run_cpl_command!("build-image", "-a", app)
+    end
+
+    app
+  end
+
+  def run_cpl_command(*args, raise_errors: false) # rubocop:disable Metrics/MethodLength
+    write_command_to_log(args.join(" "))
+
+    result = {
+      status: 0,
+      stderr: "",
+      stdout: ""
+    }
+
+    original_stderr = replace_stderr
+    original_stdout = replace_stdout
+
+    begin
+      Cpl::Cli.start(args)
+    rescue SystemExit => e
+      result[:status] = e.status
+    end
+
+    result[:stderr] = restore_stderr(original_stderr)
+    result[:stdout] = restore_stdout(original_stdout)
+
+    write_command_result_to_log(result)
+
+    raise result.to_json if result[:status].nonzero? && raise_errors
+
+    result
+  end
+
+  def run_cpl_command!(*args)
+    run_cpl_command(*args, raise_errors: true)
+  end
+
+  def spawn_cpl_command(*args, stty_rows: nil, stty_cols: nil, wait_for_process: true)
+    cmd = ""
+    cmd += "stty rows #{stty_rows} && " if stty_rows
+    cmd += "stty cols #{stty_cols} && " if stty_cols
+    cmd += "#{cpl_executable_with_simplecov} #{args.join(' ')}"
+
+    write_command_to_log(cmd)
+
+    PTY.spawn(cmd) do |output, input, pid|
+      yield(SpawnedCommand.new(output, input, pid))
+    ensure
+      Process.wait(pid) if wait_for_process
+    end
+  end
+
+  def write_command_to_log(cmd)
+    File.open(LOG_FILE, "a") do |file|
+      file.puts(command_separator)
+      file.puts(cmd)
+    end
+  end
+
+  def write_command_result_to_log(result) # rubocop:disable Metrics/MethodLength
+    File.open(LOG_FILE, "a") do |file|
+      file.puts(section_separator)
+      file.puts("STATUS: #{result[:status]}")
+      file.puts(section_separator)
+      file.puts("STDERR:")
+      file.puts(section_separator)
+      file.puts(result[:stderr])
+      file.puts(section_separator)
+      file.puts("STDOUT:")
+      file.puts(section_separator)
+      file.puts(result[:stdout])
+    end
+  end
+
+  def command_separator
```

Comment:
should this be consts?

Reply:
Done in b8a4dac2dd589ed2775d780407f253bb304e2c56.